packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -266,6 +267,77 @@ module.exports = ({ strapi }) => ({
     return mappedEntity;
   },
 
+  async publishMany(opts, uid) {
```

Comment:
wdy if we pass the entities to update directly, instead of the query to find them? (Same as with the publish)
```suggestion
  async publishMany(entities, uid) {
```

Reply:
Maby it would be better to use updateMany under the hood and do this on the entity-service layer?
Since how else are you planning to get this working with i18n?

Reply:
@Marc-Roig you mean pass them from the frontend right? Yeah I could try that. 

@Boegie19 

I am using strapi.db.updateMany here in the entity manager. 

Originally I was doing this in the entity service but I don't think these methods should be exposed there. It looks like the entities service's main responsibility is around CRUD operations and publish and unpublish don't exist there. So it didn't make sense anymore to me to have publishMany and unpublishMany in there.

> Since i18n only shows in the page view the main entity and the the other locals
I'm not sure I completely understand but you can only publish within the active locale. Meaning a bulk publish will not publish any translated versions you will have to publish them separately.

> still this means that passing the entities dirrectly would be bad since that would break warpParams
I'm not using wrapParams so maybe it will be possible... To be honest I don't think I understand what wrapParams does. It seems to just return whatever isit seems to just return whatever is passed to it but maybe either of you could explain how it works.

Reply:
I would agree on the query part , where we can pass the id list, but we need to the entity-manager service for now, to send the webhook events.

And we could use the entityService.updateMany here to be aligned with the rest of code, if it existed 😄 but we don't have bulk updates available in the entityService.

My opinion, as this is such an atomic operation (change one field) I think it is ok to use the db.query layer here.